pkgs/applications/networking/cluster/waagent/default.nix

```diff
@@ -10,33 +10,47 @@
   openssl,
   parted,
   procps, # for pidof,
-  python39, # the latest python version that waagent test against according to https://github.com/Azure/WALinuxAgent/blob/28345a55f9b21dae89472111635fd6e41809d958/.github/workflows/ci_pr.yml#L75
+  python39, # the latest python version that waagent tests against, according to https://github.com/Azure/WALinuxAgent/blob/28345a55f9b21dae89472111635fd6e41809d958/.github/workflows/ci_pr.yml#L75
   shadow, # for useradd, usermod
   util-linux, # for (u)mount, fdisk, sfdisk, mkswap
+  e2fsprogs,
+  bash,
+  lsof,
+  writeShellApplication,
 }:
 
 let
-  inherit (lib) makeBinPath;
+  walinuxagent = python39.pkgs.buildPythonPackage rec {
```

Comment:
Why moving this into some let binding? Can't you just keep the expression as-is, and only add bash/lsof/e2fsprogs to runtimeDeps?

Reply:
I think doing it this way makes it difficult to override the walinuxagent package attributes because the final output here comes from writeShellApplication. I prefer the old wrapProgram approach.

Reply:
There's no need for a separate writeShellApplication derivation, and turns out, wrapping the binary isn't needed at all. See https://github.com/NixOS/nixpkgs/pull/274220.